packages/optimizer/README.md

```diff
@@ -170,6 +170,23 @@ Automatically import any missing AMP Extensions (e.g. amp-carousel). This is not
 - default: `false`
 - used by: [AutoExtensionImport](lib/transformers/AutoExtensionImporter.js)
 
+#### `cache`
+
+Specifies the catch implementation to use for caching artifacts required during transformation (runtime CSS, validation rules, latest runtime version,...). A cache needs to conform to the following interface:
+
+```typescript
+interface Cache {
+  set(key: Object, value: Object): void;
+  get(key: Object): Object;
+}
```

Comment:
This interface guarantees you'll always get an `Object` back.

What about the case where you want to retrieve something from cache that was not yet cached or can't be fetched?

I usually see two different conventions:
- The cache can return `null`, `false` or similar to denote that nothing could be retrieved. This goes counter to strong typing.
- The cache throws an error/exception if nothing could be retrieved. This still enforces strong typing. However, in that case, you usually have a `has()` method as well to be able to avoid the error condition upfront.

If no `has()` is provided, there should be some documentation (docblock?) that defines how the error condition can be detected.